Corona-Warn-App/src/main/java/de/rki/coronawarnapp/gstatus/notification/GStatusNotificationService.kt

```diff
@@ -0,0 +1,48 @@
+package de.rki.coronawarnapp.gstatus.notification
+
+import de.rki.coronawarnapp.ccl.dccwalletinfo.model.DccWalletInfo
+import de.rki.coronawarnapp.ccl.dccwalletinfo.notification.DccWalletInfoNotificationService
+import de.rki.coronawarnapp.covidcertificate.common.certificate.CertificatePersonIdentifier
+import de.rki.coronawarnapp.covidcertificate.notification.PersonNotificationSender
+import de.rki.coronawarnapp.covidcertificate.person.core.PersonCertificatesSettings
+import de.rki.coronawarnapp.tag
+import timber.log.Timber
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class GStatusNotificationService @Inject constructor(
+    private val personNotificationSender: PersonNotificationSender,
+    private val personCertificatesSettings: PersonCertificatesSettings
+) : DccWalletInfoNotificationService {
+
+    override suspend fun notifyIfNecessary(
+        personIdentifier: CertificatePersonIdentifier,
+        oldWalletInfo: DccWalletInfo?,
+        newWalletInfo: DccWalletInfo
+    ) {
+        Timber.tag(TAG).v("notifyIfNecessary() - Started")
+        val oldGStatusId = oldWalletInfo?.admissionState?.identifier
+        val newGStatusId = newWalletInfo.admissionState.identifier
+        when {
+            newGStatusId != null && newGStatusId != oldGStatusId -> {
```

Comment:
We could have a G-status (already calculated ones) ,but no `oldGStatusId` and in this case it will a notification,since the `newGStatusId` will have a value , is that what is required or only when there is an actual change in the values and they exist in both old and new G-status?

Reply:
The tech spec only says to issue the notification and badge if the id existst and is different from the old value, but I think you're right. The old value must not be null either, otherwise you also get the badge when you first scan a certificate.